homeassistant/components/fritzbox_callmonitor/config_flow.py

```diff
@@ -0,0 +1,233 @@
+
+            result = await self.hass.async_add_executor_job(self._try_connect)
+
+            if result == RESULT_SUCCESS:
+                if len(self._phonebook_ids) > 1:
+                    return await self.async_step_phonebook()
+
+                self._phonebook_id = DEFAULT_PHONEBOOK
+                self._phonebook_name = await self._get_name_of_phonebook(
+                    self._phonebook_id
+                )
+
+                if self._is_already_configured(self._host, self._phonebook_id):
+                    return self.async_abort(reason="already_configured")
+
+                return self._get_entry()
+            if result != RESULT_INVALID_AUTH:
+                return self.async_abort(reason=result)
+            errors["base"] = result
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA_USER, errors=errors
+        )
+
+    async def async_step_phonebook(self, user_input=None):
+        """Handle a flow to chose one of multiple available phonebooks."""
+        errors = {}
+        phonebooks = []
+
+        for phonebook_id in self._phonebook_ids:
+            phonebooks.append(await self._get_name_of_phonebook(phonebook_id))
+
+        if user_input is not None:
+            phonebook_name = user_input[CONF_PHONEBOOK]
+            phonebook_id = phonebooks.index(phonebook_name)
+
+            self._phonebook_name = phonebook_name
+            self._phonebook_id = phonebook_id
+
+            if self._is_already_configured(self._host, self._phonebook_id):
+                return self.async_abort(reason="already_configured")
+
+            return self._get_entry()
+
+        return self.async_show_form(
+            step_id="phonebook",
+            data_schema=vol.Schema({vol.Required(CONF_PHONEBOOK): vol.In(phonebooks)}),
+            errors=errors,
+        )
+
+
+class FritzBoxCallMonitorOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle a fritzbox_callmonitor options flow."""
+
+    def __init__(self, config_entry):
+        """Initialize."""
+        self.config_entry = config_entry
+        self._prefixes = None
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        if user_input is not None:
+
+            self._prefixes = user_input.get(CONF_PREFIXES)
+
+            if self._prefixes and self._prefixes.strip():
+                self._prefixes = [
+                    prefix.strip() for prefix in self._prefixes.split(",")
+                ]
+
+            return self.async_create_entry(
+                title="", data={CONF_PREFIXES: self._prefixes}
+            )
+
+        return self.async_show_form(
+            step_id="init",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_PREFIXES,
+                        default=self.config_entry.options.get(CONF_PREFIXES),
+                    ): str,
```

Comment:
This does not work completely, because prefixes can be added but after that you can never go back to having no prefixes. Is there a way around that issue? 

Also I have taken a look at config_validator.py and have found ensure_csv_list but could not get it to work here. Can't these functions be used inside of a config flow schema?

Reply:
Solved in https://github.com/home-assistant/core/pull/40736/commits/6d1f82f86323f605241afefa8bb6e0ff4d77bca2 via `suggested_value`